habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py

```diff
@@ -633,8 +639,10 @@ def add_cat_fields_to_episodes(
         episode["candidate_goal_receps"] = get_candidate_receptacles(
             all_rec_goals, goal_rec_cat
         )
-        if enable_add_viewpoints and not add_viewpoints(
-            sim, episode, obj_idx_to_name, debug_viz
+        if (
+            enable_add_viewpoints
+            and not add_viewpoints(sim, episode, obj_idx_to_name, debug_viz)
+            or start_rec_cat == goal_rec_cat
```

Comment:
Minor: we could move this check a bit ahead to save some time.

Reply:
I agree this check should just be moved earlier to clean up this if statement and avoid calling `add_viewpoints`.